IdentityCore/src/requests/MSIDInteractiveAuthorizationCodeRequest.m

```diff
@@ -232,16 +139,97 @@ - (void)showWebComponentWithCompletion:(MSIDWebviewAuthCompletionHandler)complet
 {
-    MSIDAuthorizationCodeResult *result = [MSIDAuthorizationCodeResult new];
-    result.authCode = authCode;
-    result.accountIdentifier = self.authCodeClientInfo.accountIdentifier;
-    result.pkceVerifier = self.webViewConfiguration.pkce.codeVerifier;
-    completionBlock(result, nil, nil);
+    void (^returnErrorBlock)(NSError *) = ^(NSError *localError)
+    {
+        NSString *errorString = [localError msidServerTelemetryErrorString];
+        if (errorString)
+        {
+#if !EXCLUDE_FROM_MSALCPP
+            [self.lastRequestTelemetry updateWithApiId:[self.requestParameters.telemetryApiId integerValue]
+                                           errorString:errorString
+                                               context:self.requestParameters];
+#endif
+        }
+        
+        completionBlock(nil, localError, nil);
+    };
+    
+    if (error)
+    {
+        returnErrorBlock(error);
+        return;
+    }
+
+    if ([response isKindOfClass:MSIDWebOpenBrowserResponse.class])
+    {
+        error = nil;
+        MSIDWebResponseBaseOperation *operation = [MSIDWebResponseOperationFactory createOperationForResponse:response
+                                                                                                        error:&error];
+        if (error)
+        {
+            returnErrorBlock(error);
+            return;
+        }
+        
+        BOOL isCurrentFlowFinished = [operation doActionWithCorrelationId:self.requestParameters.correlationId
+                                                                    error:&error];
+        if (isCurrentFlowFinished && error)
+        {
+            returnErrorBlock(error);
+            return;
+        }
+        
+        // This should never happen, create a new error here just in case it would hang if somehow falls into this part
+        error = MSIDCreateError(MSIDErrorDomain,
+                                MSIDErrorInternal,
+                                @"Authorization session was not canceled successfully",
+                                nil,
+                                nil,
+                                nil,
+                                self.requestParameters.correlationId,
+                                nil,
+                                YES);
+        returnErrorBlock(error);
+        return;
+    }
+        
+    NSError *localError = nil;
+    MSIDWebResponseBaseOperation *operation = [MSIDWebResponseOperationFactory createOperationForResponse:response
+                                                                                                    error:&localError];
+    
+    if (localError)
+    {
+        returnErrorBlock(localError);
+        return;
+    }
+
+    __typeof__(self) __weak weakSelf = self;
+    [operation invokeWithRequestParameters:self.requestParameters
```

Comment:
Should this new refactored logic be guarded with a feature flag in case a scenario is missed and breaks?